test/unit/rules/lint-no-childless-elements-test.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+generateRuleTests({
+  name: 'no-childless-elements',
+
+  config: true,
+
+  good: [
+    '<h1>something</h1>',
+    '<h2>something</h2>',
+    '<h3>something</h3>',
+    '<h4>something</h4>',
+    '<h5>something</h5>',
+    '<h6>something</h6>',
+    '<p>something</p>',
+    '<span>something</span>',
+    '<a>something</a>',
+    '<ul>something</ul>',
+    '<ol>something</ol>',
+    '<li>something</li>',
+    '<div>something</div>',
+  ],
+
+  bad: [
+    {
+      template: '<h1></h1>',
+      message: 'Empty non-void elements are not allowed',
```

Comment:
The error message should be imported from the rule file. See other rules like `no-negated-condition` for an example.